### ISSUE TITLE
A detailed proposal to completely rework ground battle system, to bring a more dynamic campaign

### ISSUE THREAD

**Issue by hymlee** [dcs-liberation/dcs_liberation#2227]
**Foreword and Current Shortcomings**
The current ground battle system features only simple forward(aggression) and backward(retreat) movement. In reality, the battlespace is 2-dimensional, where troops can move in any direction, executing different tactics such as flanking. Under the current 1-dimensional forward and backward logic, no strategy can be implemented. In addition, the current battle system focuses on battles between heavy tanks, with little to none emphasis on the importance of light units. I understand my idea may not be perfect. So I really appreciate help from all of you to make DCS liberation even better.

**Suggestions**
To introduce a 2 dimensional battlespace, we can make reference to how sea battles are constructed. In the current sea battle system, there is no front line. Ships are grouped into groups and each group of ships arrive can freely move around at sea for certain range in a round. We can copy such model and add a few enhancememts, which I will describe below in details.

- Remove frontline and connections between capture points.

- Units are grouped into "Divisions".
Division can purchase/reinforce units.

- Divisions, based on their types(armored/infantry/Calvary/AA battery) and unit composition, have different scout distance which they can spot and unveil enemy within certain radius, and have different travelling distance per round.

- Divisions can freely move around the map in any direction. Division can also choose to face against certain direction to maximize combat effectiveness.

- There is a population cap, say 60, for each division for game balance purpose. For example, Humvee and other light vehicles take 1 population, IFV/APC takes 2 population, tank takes 3 population. So players have to carefully plan the composition of unit. 

- Additional units up to a certain population can be purchased as a reserve unit, which will replenish any destroyed unit.

- AA battery/antiship battery is no longer located and fixed in a predetermined position. Players can purchase AA/antiship division and set it up wherever it is needed.

- Buildings that have different purposes spread mostly inside city/urban area, so as to place emphasis on the importance of urban warface which will primarily be conducted by infantry division with huge amount of light units.

- Fuel depot affects maximum travelling range and scout range of each division. Each fuel depot captured provides 0.1nm increase in maximum travelling range and scouting range.

- Ammo depot affects population cap of each division. Each ammo depot captured provides 1 increase in population cap of every division.

- Base travelling distance is calculated using weighted average of unit composition.
For example, humvee has a base travel distance of 60nm; APC has a base travel distance of 45nm; tanks/IFV has a base travel distance of 30nm.
For a division comprises of 10% humvee; 40%APC, 50%tanks, travelling distance is calculated as 0.1＊60+0.4＊45+0.5＊30=39nm

- Base scouting distance is calculated using weighted average of unit composition.
For example, humvee has a base scouting distance of 35nm; APC has a base scouting distance of 35nm; IFV has a base scouting distance of 80nm; tanks has a base scouting distance of 20nm.
For a division comprises of 10% humvee; 40%IFV, 50%tanks, travelling distance is calculated as 0.13＊5+0.4＊80+0.5＊20=45.5nm

- Set a maximum number of division that can be generated, say 15. to avoid players/AI spamming with huge numbers of divisions and destroying game balance. Recommend a composition of different types of division in the entire army as follows for AI to follow, though it is freely adjustable by players.
Infantry division:30%
Armored division:30%
Calvary division: 20%
AA battery + antiship division:20%

- For each type of division, recommend a default unit composition for AI to follow, though it is freely adjustable by players.
Infantry division:30%humvee, 40%humvee ATGM, 10% IFV, 5% tanks, 5%shorad, 10%artillery
Armored division: 30%IFV, 60%tanks, 10%shorad
Calvary/reconnaissance division: 70%IFV, 20%tanks, 10%shorad

**Mission planning**

- Type of divisions is definable, which different objectives are assumed.
Infantry division- aim to capture and defend urban/city area which consist of a lot of valuable buildings.
Armored division- head for open terrain and avoid city area

- Map creator can categorize different area of the map into city/open terrain. Infantry Division will head for city area to capture and defend valuable facilities, whereas armored division will head for open terrain to search and destroy opponent's armored division.

- Each city area is considered a capture point, just like an airfield. Eliminating all enemy troops inside a certain radius around the city area represents a capture of all valuable facilities inside that city.

**AI behaviour**

- Infantry division :
When enemy is inside the same city area , the division will station inside the city to defend.
When there is no enemy inside the city, the division will depart and head for the nearest city that is under control by the enemy.

- Armored division:
(1) Advance ahead of infantry division in the open terrain to deter opponent's armored division from doing harm on self's infantry division.
(2) When encountered enemy unit and is outgunned, the division will adopt defensive stance, which the division will stop advancing and line up the unit in line abreast formation facing the enemy to provide maximum fire power. In the meantime, nearest friendly armored division will join up to reinforce. Aggressive CAS sorties will also be automatically planned with TOT set to ASAP.
(3) When encountered enemy unit and in advantageous position, division will advance forward until just inside firing range, which by then the unit will stop and engage.
(4) When in clearly victorious position, division will breakthrough to close distance and elimimate the enemy.

- Calvary division:
Advance ahead of armored division to provide scouting to unveil enemy's position.
Divisions are spaced to maximize scouting envelope.
When encountered enemy, always retreat unless significantly outgun the opponent. Nearest armoured division will reroute and engage the enemy.

This is how the revised ground battle UI could look like.

![New-Document-1](https://user-images.githubusercontent.com/105715207/170945065-77c007d3-e6d2-452b-9e5e-0d3288cc5d88.jpg)

![New-Document-2](https://user-images.githubusercontent.com/105715207/170945109-61047d8c-a801-4f3a-8ad6-5f4a9da7baba.jpg)

I dont mind taking critics, I just want to make DCS liberation even better.
So please feel free to share your opinions on my thought.

**Comment by DanAlbert** [comment]
Are you planning on providing patches?

**Comment by hymlee** [comment]
No I am not a coder. I'd love to code it myself but I just couldnt do it. Just a thought of mine for discussion.

**Comment by Starfire13** [comment]
I can see you've put a considerable amount of time into this.

One thing I'd like to point out is that there is no sea battle system. Carriers can be moved so that they can stay in range of moving frontlines. The system was not implemented with the intent of bringing surface ships into engagement range for naval battles.

**Comment by hymlee** [comment]
> I can see you've put a considerable amount of time into this.
> 
> One thing I'd like to point out is that there is no sea battle system. Carriers can be moved so that they can stay in range of moving frontlines. The system was not implemented with the intent of bringing surface ships into engagement range for naval battles.

Yes. But perhaps we can adopt a similar methodology in moving ground units around the battlefield, instead of mandating ground units fighting along one restricted path. That can unlock immense potential for such a large theater. 

We've already had a system for aircraft carriers to roam freely at sea, though manual control from player is required. I believe with some further coding, the AI could be taught to automatically steer the Divisions and bring itself into battles.

**Comment by Goliathus-21** [comment]
I´d like to point out this enhancement to be considered in reworking the ground battle system:

[https://github.com/dcs-liberation/dcs_liberation/issues/655]

There are campaigns, where you have almost no precision A/G strikers for DEAD missions (e.g. cold war era) and it would be great if the artillery could be assigned to attack the SAM defense, if it is in its range. This would make missions more interesting and any missions in the area could be postponed until the SAM defense will get softened by the artillery strikes.

At the moment the artillery usually fires at a certain point on the map, but its pretty common that all (moving) ground units are already elsewhere, so there is only a limited effect on the target. But if the SAM location is known and revealed, the player could be empowered to assign a certain enemy ground target for the artillery.

Thx. 

Goliathus
